lecture_1/homework/app/controllers/samurais_controller.rb

```diff
@@ -0,0 +1,60 @@
+class SamuraisController < ApplicationController
+  # GET /samurais
+  def index
+    samurais = clan.samurais.all()
+    render json: samurais.to_json()
+  end
+
+  # GET /samurais/1
+  def show
+    render json: samurai.to_json()
+  end
+
+  # POST /samurais
+  def create
+    samurai = clan.samurais.new(samurai_params)
+
+    if samurai.save
+      render json: samurai, status: :created
+    else
+      render json: samurai.errors, status: 422
+    end
+  end
+
+  # PATCH/PUT /samurais/1
+  def update
+    if samurai.update(samurai_params)
+      render json: samurai
+    else
+      render json: samurai.errors, status: 422
+    end
+  end
+
+  # DELETE /samurais/1
+  def destroy
+    samurai.destroy!
+  end
+
+  def alive
```

Comment:
Najlepiej nie tworzyć takich `nierestowych` akcji, tylko obsługiwać je jakoś w odpowiedniej restowej akcji(dostępne w CRUD to show, index, create, update, destroy) albo wydzielić do nowego zagnieżdżonego kontrolera, w naszym przypadku 100% full super restowo ścieżki by wyglądały tak:

```ruby
resources :clans, param: :name  do
  resources :samurais do
    resources :alive, only: %i[index]
    resources :dead, only: %i[index]
  end
end
```

Oczywiście staramy się unikać zagnieżdżeń więc to póki co samo obsługiwanie 2 query params w indexie by wystarczyło, ale jeśli mielibyśmy np. inne akcje `update` dla żywych i martwych samurai, to taką architekturę dużo łatwiej się rozbudowuje i ładniej dzieli tą logikę :D